File: CampaignManagerAPI/Models/Permission.cs
using Lib.Enums;
using System;
using System.Collections.Generic;
using System.Text;

namespace API.Models
{
    public class Permission
    {
        public PermissionType PermissionType { get; set; }
        public Identity Grantor { get; set; }
        public Group Grantee { get; set; }
    }
}

File: FrontEnd/src/app/interfaces/permission.ts
import { Identity } from './identity';
import { Group } from './group';

export enum PermissionType{
    Owner = 0,
    Editor = 1,
    Viewer = 2
}

export interface Permission{
    permissionType: PermissionType;
    grantor: Identity;
    grantee: Group;
}